### PR TITLE
feat(tui): add FilterBar to planned operations (#21 PR3)

### PR DESCRIPTION
**Pull request by corentin-core**
## Summary

- Add `FilterBar` (search + category) to the planned operations table
- Local filtering logic in `PlannedOperationsWidget` with `_filter_operations()` static method
- Split `_operations` into `_all_operations` / `_filtered_operations` for clean state management

## Changes

- `budget_forecaster/tui/screens/planned_operations.py` — FilterBar integration, filter handlers, local filtering
- `tests/tui/test_planned_operations_widget.py` — 10 tests covering search, category, combined, reset, edge cases

## Test plan

- [x] All 721 tests pass
- [x] Pre-commit hooks pass (black, mypy, pylint, ruff)
- [x] FilterBar shows search + category (no date/amount for planned ops)
- [x] Search filters by description (case-insensitive)
- [x] Category filter works
- [x] Combined filters work
- [x] Reset clears filters

Part of #21 — builds on PR2 (#233)

### CODE REVIEW

File: tests/tui/test_planned_operations_widget.py [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+
+class TestPlannedOperationsWidgetFiltering:
+    """Integration tests for PlannedOperationsWidget filtering."""
+
+    async def test_all_operations_shown_initially(self) -> None:
+        """All planned operations are displayed when no filter is applied."""
+        app = PlannedOpsWidgetTestApp()
+        async with app.run_test():
+            table = app.query_one("#planned-ops-table", DataTable)
+            assert table.row_count == 5
```
[the remark this answers is not in the document]
Comment (corentin-core) [reply tests/tui/test_planned_operations_widget.py]: 🤖 Claude: Fixed in 5d11b3f. Added public `planned_operations` property on `PlannedOperationsWidget`, all filter tests now assert on the set of descriptions instead of just the row count.

File: tests/tui/test_planned_operations_widget.py [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+
+    async def test_filter_by_category(self) -> None:
+        """Filtering by category shows only matching operations."""
+        app = PlannedOpsWidgetTestApp()
+        async with app.run_test() as pilot:
+            app.query_one("#filter-category", Select).value = Category.RENT.name
+            await pilot.click("#filter-apply")
+
+            table = app.query_one("#planned-ops-table", DataTable)
+            assert table.row_count == 2
```
[the remark this answers is not in the document]
Comment (corentin-core) [reply tests/tui/test_planned_operations_widget.py]: 🤖 Claude: Fixed in 5d11b3f.

File: tests/tui/test_planned_operations_widget.py [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+    async def test_filter_bar_shows_filtered_count(self) -> None:
+        """Filter bar status shows filtered vs total count."""
+        app = PlannedOpsWidgetTestApp()
+        async with app.run_test() as pilot:
+            app.query_one("#filter-search", Input).value = "loyer"
+            await pilot.click("#filter-apply")
+
+            status = app.query_one("#filter-status", Static)
+            rendered = str(status.render())
+            assert "2" in rendered
```
[the remark this answers is not in the document]
Comment (corentin-core) [reply tests/tui/test_planned_operations_widget.py]: 🤖 Claude: Fixed in 5d11b3f.
